test/helper/test_timeline.ts

```diff
@@ -19,6 +25,233 @@ const parseTimelineFileFromTriggerFile = (filepath: string) => {
+    lineNumber: number,
+  ): void {
+    const origLine = line;
+    // First, reduce all double-quoted strings to just "".  We don't check/lint string contents,
+    // and this avoids various problems like inadvertently matching double-spaces inside a string,
+    // or the use of a {} regex occurence modifier causing errors in identifying line groups.
+    line = line.replace(/"[^"]*?"/g, '""');
+
+    line = line.replace(/\r/g, ''); // remove \r before checking for extraneous whitespace
+    if (line.trim() !== line) {
+      this.lintErrors.push({
+        lineNumber: lineNumber,
+        line: origLine,
+        error: 'Line has leading or trailing whitespace',
+      });
+    }
+    line = line.trim();
+
+    // Remove in-line comments from further lint checks
+    line = line.replace(/#.*$/, '').trim();
+    if (line.length === 0)
+      return;
+
+    // There should be no remaining allowable double-spacing within the line
+    if (line.match(/ {2}.+/)) {
+      this.lintErrors.push({
+        lineNumber: lineNumber,
+        line: origLine,
+        error: 'Line has double spaces not enclosed by quotes',
+      });
+      // don't return; continue processing the line
+    }
+
+    // Capture each part of the line (separated by spaces).
+    // Anything encapsulated by double-quotes or braces will be treated as a single element.
+    const lineParts = line.match(/"[^"]*"|\{[^}]*\}|[^ ]+/g);
+    if (lineParts === null || lineParts[0] === undefined) {
+      this.lintErrors.push({
+        lineNumber: lineNumber,
+        line: origLine,
+        error: 'Bad format - cannot parse line into parts for further linting',
+      });
+      return;
+    }
+
+    const first = lineParts[0];
+    if (first === 'hideall') {
+      // parse() throws errors if a hideall line has an invalid format
+      return;
+    }
+
+    // At this point, if `first` is not a time, it's not a valid timeline entry
+    const time = parseFloat(first);
+    if (isNaN(time)) {
+      this.lintErrors.push({
+        lineNumber: lineNumber,
+        line: origLine,
+        error: 'Line does not begin with a config keyword, comment, or sync time.',
+      });
+      return;
+    }
+
+    // Ensure that the time is either an integer or has a single digit after the decimal
+    if (!Number.isInteger(time) && !/^\d+\.\d$/.test(time.toString())) {
+      this.lintErrors.push({
+        lineNumber: lineNumber,
+        line: origLine,
+        error: 'Sync time must be an integer or float with a single-digit fractional',
+      });
+      // don't return; continue processing the line
+    }
+
+    // Enforce chronological order of sync lines
+    if (!this.ignoreSyncOrder) {
+      if (time < this.lastSyncTime)
+        this.lintErrors.push({
+          lineNumber: lineNumber,
+          line: origLine,
+          error:
+            `Sync time of "${time.toString()}" predates prior entry of "${this.lastSyncTime.toString()}"`,
+        });
+      this.lastSyncTime = time;
```

Comment:
This looks like it could potentially make the current time position move backward. Is that intentional?

Reply:
> This looks like it could potentially make the current time position move backward. Is that intentional?

No, it's not - that's an unfortunate bug I didn't catch when I rewrote that part of the code to simplify the logic.